src/design-pages/idoverride_redesign.rst

```diff
@@ -0,0 +1,400 @@
+same is true for group id-overrides with respect to the GID. As seen above the
+attribute for the primary GID of the user does not have to be unique and if is
+not an attribute we search for. Currently we use the same attribute name
+'gidNumber' for both but this should be avoided to not bloat the index for the
+group related attribute, which is expected to be unique, with the primary
+groups of users. An attribute name like 'userGidNumber' should help to avoid
+this.
+
+Certificates are not expected to be unique in user id-overrides. This supports
+use-cases where a single certificate (a single Smartcard) can be used to log
+into different accounts. Think, e.g. of temporary/holiday replacements or
+multiple accounts of a single physical user for different purposes. So we
+cannot avoid indexes with multiple entries in this case. However it is
+expected that for the above use-cases the multiple use of a single certificate
+is limited to small numbers in the order of 10 where adding new entries to the
+index should not cause a considerable slowdown. Nevertheless it might be worth
+to mention this in the documentation/man page.
+
+An potential additional optimization might be to use indexed attributes with
+names with a dedicated prefix, e.g. 'ido' for id-override. With this a search
+filter for an user id-override with a given UID can be just 'idoUidNumber=UID'
+and the object can be directly returned from the indexed search without
+evaluating additional attributes. But since the additional evaluation would
+just be a string comparison I would expect only a limited speedup, if any.
+
+Managing id-overrides in SSSD's cache
+=====================================
+Currently there are two types of sources for the id-override object. They are
+either added locally with the 'sss_override' command or read from LDAP. With
+respect to LDAP currently only IPA is supported properly, but with this
+re-design adding support in plain LDAP or AD should be more easy.
+
+Local overrides
+---------------
+The 'sss_override' command should just add or modify the id-override object
+without trying to read the original object. This should speed up adding
+multiple id-overrides in a row and would also allow to add the id-overrides
+while the system is offline, e.g. when creating images or provisioning a
+system in a special environment.
+
+This separation also allows to keep the id-override in the cache while the
+original object is deleted from the cache. The primary purpose here is to
+avoid to loose the id-override if the original object was removed from the
+cache due to an error in SSSD, on the server side or in the configuration. The
+drawback is that the id-override becomes useless if the original object was
+removed on the serve side. But this issue is imo easier to handle than an
+id-override that was lost accidentally.
+
+The 'sss_override' command currently writes the id-override objects directly
```

Comment:
Maybe `sss_override` could be modified to identify, list and delete dead overrides.